crates/deltalake-core/src/lib.rs

```diff
@@ -77,7 +77,11 @@ compile_error!(
     "Features parquet and parquet2 are mutually exclusive and cannot be enabled together"
 );
 
-#[cfg(all(feature = "s3", feature = "s3-native-tls"))]
+#[cfg(all(
```

Comment:
The correct way should be like `#[cfg(any(all(A, B), all(B, C), all(A, C)))]` so that any two of them cannot be enabled together. But it does not affect our usage.